truss/truss_config.py

```diff
@@ -372,3 +347,50 @@ def clone(self):
     def validate(self):
         for secret_name in self.secrets:
             validate_secret_name(secret_name)
+
+
+DATACLASS_TO_REQ_KEYS_MAP = {
+    Train: {"variables"},
+    Resources: {"accelerator", "cpu", "memory", "use_gpu"},
+    TrussConfig: {
+        "environment_variables",
+        "external_package_dirs",
+        "model_metadata",
+        "model_name",
+        "python_version",
+        "requirements",
+        "resources",
+        "secrets",
+        "system_packages",
+    },
+    BaseImage: {"image", "python_executable_path"},
+}
+
+
+def obj_to_dict(obj):
```

Comment:
please add a doc string or inline comments so we can more easily parse this when we looks to extend/change the config class